internal/pkg/agent/application/paths/paths_unix.go

```diff
@@ -16,3 +17,28 @@ func initialControlSocketPath(topPath string) string {
 
 // ResolveControlSocket does nothing on non-Windows hosts.
 func ResolveControlSocket() {}
+
+// HasPrefix tests if the path starts with the prefix.
+func HasPrefix(path string, prefix string) bool {
```

Comment:
Depending on where the current directory is when we do this check, the evaluation could be done using [filepat.IsLocal](https://pkg.go.dev/path/filepath#IsLocal)

Reply:
maybe, but I found enough edge cases with the current algorithm that I'm reluctant to change .  FYI this implementation is very similar to https://github.com/golang/dep/pull/395, which had a fair amount of users.